sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py

```diff
@@ -122,7 +148,7 @@ class TextTranslationClient(ServiceClientGenerated):
          https://api.cognitive.microsofttranslator.com). Required.
     :type endpoint: str
     :param credential: Credential used to authenticate with the Translator service
-    :type credential: Union[AzureKeyCredential , TokenCredential , TranslatorCredential]
+    :type credential: Union[AzureKeyCredential , TokenCredential , TranslatorCredential, TranslatorAADCredential]
```

Comment:
Can we instead support the TokenCredentials provided by the azure-identity package? If we do this, then you should be able to get all of that goodness and credential logic without having to maintain it in this library.

Reply:
Also, if you now support AAD auth in your service, then you should specify that in the TypeSpec as well

Reply:
I am not sure what do you mean and what does it provides but I would guess it doesn't provide the two MT specific headers that we need when global MT endpoint is used. https://learn.microsoft.com/en-us/azure/ai-services/translator/reference/v3-0-reference#authentication-with-microsoft-entra-id

Reply:
@xiangyan99 @pvaneck do we have support for this auth scenario in azure identity?

Reply:
@MikeyMCZ,  do you use same or different scopes/audiences for global endpoint vs custom endpoint?

Reply:
@catalinaperalta if I understand correctly, here we still want to use token credentials. The trick is in addition to the token itself, for global endpoint scenarios, we also need the information of `Resource ID` and `region`

Reply:
Btw, you may need to discuss with architects about 
1. If we need two different `TranslatorAADCredential` classes?
2. The field names of the credential class. To be clearer:
        self.tokenCredential = tokenCredential
        self.resourceId = resourceId
        self.region = region
3. Given these different combinations (endpoint + credentials), do we want to use overloads?

Reply:
_do you use same or different scopes/audiences for global endpoint vs custom endpoint?_
We are using the same scope for the token.

_if I understand correctly, here we still want to use token credentials. The trick is in addition to the token itself, for global endpoint scenarios, we also need the information of Resource ID and region_
That is correct. Same change was already accepted in dotnet and Java should be approved and merge soon.

Reply:
_Given these different combinations (endpoint + credentials), do we want to use overloads?_
I am ok with any change that will be advised - I am not sure what is the best practice in the Python. Do you mind providing the code suggestion that I can accept?